src/express-server.ts

```diff
@@ -0,0 +1,229 @@
+): StreamableHTTPServerTransport {
+  const httpTransport = httpTransportMap.get(sessionId);
+  if (!httpTransport) {
+    console.error(`No transport found for sessionId: ${sessionId}`);
+    throw new HTTPError(400, ERROR_MESSAGES.TRANSPORT_NOT_FOUND);
+  }
+  return httpTransport;
+}
+
+/**
+ * Creates a new transport and connects it to the MCP server
+ */
+async function createAndConnectTransport(
+  mcpServer: McpServer,
+  req: Request,
+  res: Response,
+): Promise<void> {
+  const httpTransport = new StreamableHTTPServerTransport({
+    sessionIdGenerator: () => randomUUID(),
+  });
+
+  await mcpServer.connect(httpTransport);
+  await httpTransport.handleRequest(req, res, req.body);
+
+  const sessionId = httpTransport.sessionId;
+  if (sessionId) {
+    httpTransportMap.set(sessionId, httpTransport);
+  }
+}
+
+/**
+ * Handles existing session requests
+ */
+async function handleExistingSession(
+  sessionId: string,
+  req: Request,
+  res: Response,
+): Promise<void> {
+  const httpTransport = getTransportForSession(sessionId);
+  await httpTransport.handleRequest(req, res, req.body);
+}
+
+const app = express();
+app.use(express.json());
+
+// Middleware to validate authorization and create MCP server
+app.use((req, _res, next) => {
+  try {
+    const authHeader = validateAuthHeader(req.headers[HEADERS.AUTHORIZATION]);
+    req.mcpServer = createMCPServerInstance(authHeader);
+    next();
+  } catch (error) {
+    next(error);
+  }
+});
+
+// Error handling middleware
+app.use((err: HTTPError, _req: Request, res: Response, _next: NextFunction) => {
+  console.error("Server error:", err.message);
+  const status = err.status || 500;
+  const message = status === 500 ? ERROR_MESSAGES.INTERNAL_ERROR : err.message;
+  res.status(status).json({ error: message });
+});
+
+app.get("/mcp", async (req, res, next) => {
+  try {
+    const sessionId = validateSessionId(req.headers[HEADERS.SESSION_ID]);
+    const httpTransport = getTransportForSession(sessionId);
+    await httpTransport.handleRequest(req, res, req.body);
+    // ?TODO: this.streamMessages(httpTransport)
+  } catch (error) {
+    next(error);
+  }
+});
+
+app.post("/mcp", async (req, res, next) => {
+  try {
+    const sessionId = req.headers[HEADERS.SESSION_ID] as string | undefined;
+
+    // Handle initialization requests (no session ID required)
+    if (!sessionId && isInitializeReq(req.body)) {
+      await createAndConnectTransport(req.mcpServer, req, res);
+      return;
+    }
+
+    // Handle existing session requests
+    if (sessionId) {
+      await handleExistingSession(sessionId, req, res);
+      return;
+    }
+
+    // Invalid request - no session ID and not an initialize request
+    throw new HTTPError(400, ERROR_MESSAGES.INVALID_SESSION);
+  } catch (error) {
+    next(error);
+  }
+});
+
+app.get("/", (_req: Request, res: Response) => {
+  res.send(
+    "This is the Firehydrant MCP server. Use the /mcp endpoint for MCP connections.",
```

Comment:
We could reuse the cloudflare page ultimately

Reply:
I'm wondering if a generated, static HTML page would work well for both?